libs/common/src/platform/state/state-definition.ts

```diff
@@ -1,16 +1,57 @@
+/**
+ * Default storage location options.
+ *
+ * `disk` generally means state that is accessible between restarts of the application,
+ * with the exception of the web client. In web this means `sessionStorage`. The data is
+ * through refreshes of the page but not available once that tab is closed or from any
+ * other tabs.
+ *
+ * `memory` means that the information stored there goes away during application
+ * restarts.
+ */
 export type StorageLocation = "disk" | "memory";
 
+/**
+ * *Note*: The property names of this object should match exactly with the string values of the {@link ClientType} enum
+ */
+export type ClientLocations = {
+  /**
+   * Overriding storage location for the web client.
+   *
+   * Includes an extra storage location to store data in `localStorage`
+   * that is available from different tabs and after a tab has closed.
+   */
+  web: StorageLocation | "disk-local";
+  /**
+   * Overriding storage location for browser clients.
+   */
+  browser: StorageLocation;
+  /**
+   * Overriding storage location for desktop clients.
+   */
+  desktop: StorageLocation;
+  /**
+   * Overriding storage location for CLI clients.
+   */
+  cli: StorageLocation;
```

Comment:
Bit of a gotcha here that most of these won't work.

Let's comment them out until such a time as we need overrides here
```suggestion
  web: StorageLocation | "disk-local";
  /**
   * Overriding storage location for browser clients.
   */
  //browser: StorageLocation;
  /**
   * Overriding storage location for desktop clients.
   */
  //desktop: StorageLocation;
  /**
   * Overriding storage location for CLI clients.
   */
  //cli: StorageLocation;
```